internal/catalogmetadata/cache/cache_test.go

```diff
@@ -0,0 +1,269 @@
+package cache_test
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
+
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata/cache"
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata/client"
+)
+
+const (
+	package1 = `{
+		"schema": "olm.package",
+		"name": "fake1"
+	}`
+
+	bundle1 = `{
+		"schema": "olm.bundle",
+		"name": "fake1.v1.0.0",
+		"package": "fake1",
+		"image": "fake-image",
+		"properties": [
+			{
+				"type": "olm.package",
+				"value": {"packageName":"fake1","version":"1.0.0"}
+			}
+		]
+	}`
+
+	stableChannel = `{
+		"schema": "olm.channel",
+		"name": "stable",
+		"package": "fake1",
+		"entries": [
+			{
+				"name": "fake1.v1.0.0"
+			}
+		]
+	}`
+)
+
+func TestClient(t *testing.T) {
+	t.Run("Bundles", func(t *testing.T) {
+		type test struct {
+			name       string
+			catalog    *catalogd.Catalog
+			contents   []byte
+			wantErr    bool
+			tripper    *MockTripper
+			extraTests func(t *testing.T, c client.Fetcher, ctx context.Context, tst test)
```

Comment:
I've not seen this pattern before, and I'm worried it might be confusing, ripe for misuse later, or become an abstraction that is no longer useful and that we keep carrying because it's here.

WDYT of factoring these into tests into separate `Test*` functions or `t.Run` blocks that just do a series of things in order?

Reply:
TLDR: I'd prefer this be done in a follow-up if acceptable since this follows the table driven testing the rest of the `catalogmetadata` package seems to follow for multiple tests of the same method.

The thinking here was that this would be a function signature used to run a few extra validation/test steps and at it's base, all of these test cases have to go through the same initial run. The idea was to reduce code repetition while still achieving the ability to do some additional testing steps for the, currently, 2 tests that need it. 

I'd prefer to keep this as is for now since it follows the same table driven testing pattern that exists in the rest of the `catalogmetadata` package and reduces code repetition.

Reply:
I'm not opposed to table-driven, but that sort of hits on my concern. This isn't totally table-driven because the table actually contains more test code rather then being more declarative/static test input/expectations.

Reply:
If I'm the only one with this concern, I won't hold this PR up on it though.

Reply:
That's a fair point - I'm still new to the table driven testing pattern so I wasn't aware this was actually an anti-pattern for it. Thanks for pointing it out, til :)

Changed it in https://github.com/operator-framework/operator-controller/pull/411/commits/76c9cc8412d8a4ff191989f841802243262b8274 - let me know if this addresses your concerns